### PR TITLE
medkits: Remove hardcoded regen_max; use hp_max instead

### LINKED ISSUE #571: Medkits seem to not work with knights

### PR DESCRIPTION
**Pull request by ClobberXD**
Tested; works - players are healed fully, irrespective of their class.

Fixes #571

### CODE REVIEW

File: mods/pvp/medkits/init.lua [hunk 1]
```diff
@@ -95,12 +96,14 @@ minetest.register_globalstep(function(dtime)
...
+			if pstat then
 				local hp = player:get_hp()
-				if hp < regen_max then
-					player:set_hp(math.min(hp + regen_step, regen_max))
+				if hp < pstat.regen_max then
+					player:set_hp(math.min(hp + regen_step, pstat.regen_max))
 				else
-					player:set_hp(regen_max)
+					-- TO BE REMOVED. PUNCH ME IN THE FACE IF I FORGET TO REMOVE THIS
+					minetest.chat_send_all("Regen target reached: " .. hp .. "/" .. pstat.regen_max)
```
Comment (rubenwardy) [inline mods/pvp/medkits/init.lua:106]: remember to remove this

**Comment by rubenwardy** [comment]
looks good apart from that

File: mods/pvp/medkits/init.lua (hunk 1 above)
Comment (ClobberXD) [reply mods/pvp/medkits/init.lua:106]: shoot

Comment (ClobberXD) [reply mods/pvp/medkits/init.lua]: Sorry; done. I used that to confirm that the `set_hp(regen_max)` was redundant.

**Pull request merged**